### ISSUE TITLE
MPI-4: Session error handler function not called

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#11407]
I'm working on implementing `Create_errhandler()` methods in mpi4py. I've always procrastinate to do it because a) no one ever asked, and b) creating error handlers with the C bindings is definitely not friendly for dynamic languages like Python, so a compromise have to be made (e.g. limiting the number of error handlers that can be ever created). However, after mpi4py/mpi4py#307 from @bosilca to support ULFM and realizing that many ULFM tests/examples/tutorials do use custom error handlers, I decided to move forward in mpi4py/mpi4py#310. As usual, I've added some lightweight testing, but one of my tests failed when using `Session`, but `Comm`/`Win`/`File` are OK.

#### Reproducer
```python
from mpi4py import MPI

mpicls = MPI.Session
#mpicls = MPI.Comm

if mpicls is MPI.Session:
    mpiobj = MPI.Session.Init()
if mpicls is MPI.Comm:
    mpiobj = MPI.COMM_WORLD.Dup()

called = False
check = False

def errhandler_fn(arg, err):
    global called
    global check
    called = True
    check = True
    check &= (arg == mpiobj)
    check &= (err == MPI.ERR_OTHER)

errhandler = mpicls.Create_errhandler(errhandler_fn)
mpiobj.Set_errhandler(errhandler)
mpiobj.Call_errhandler(MPI.ERR_OTHER)
errhandler.Free()

if mpicls is MPI.Session:
    mpiobj.Finalize()
if mpicls is MPI.Comm:
    mpiobj.Free()

assert called
assert check

```
#### Expected Output
No output at all, the `assert`  statements should succeed.

#### Actual Output
This is with  a recent (yesterday) build of `ompi@main`. 
```
$ python test.py
Traceback (most recent call last):
  File "/tmp/test.py", line 32, in <module>
    assert called
AssertionError
```
NOTE: Running the reproducer with MPICH succeeded, so I don't think the problem is within the new mpi4py code in mpi4py/mpi4py#310

**Comment by dalcinl** [comment]
@hppritcha You may want to take a look at this one.

**Comment by hppritcha** [comment]
closed via #11502 and
#11538
